### PR TITLE
Recipe timeout for pre-ambari phase

### PR DESCRIPTION
**Pull request by attilapalfi92**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeEngine.java [hunk 1]
```diff
@@ -89,6 +89,14 @@ public void uploadUpscaleRecipes(Stack stack, HostGroup hostGroup, Set<HostMetad
         }
     }
 
+    public void executePreAmbariStartRecipes(Stack stack, Set<HostGroup> hostGroups) throws CloudbreakException {
+        Orchestrator orchestrator = stack.getOrchestrator();
+        if (recipesSupportedOnOrchestrator(orchestrator)
+            && (recipesFound(hostGroups) || (stack.getCluster() != null && stack.getCluster().getLdapConfig() != null))) {
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeEngine.java]: 1, The conditions aren't ordered by weight
2, Hostgroups are part of cluster, is it possible that `stack.getCluster() != null` is unnecessary?

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeExecutionFailureCollector.java [hunk 2]
```diff
@@ -22,6 +22,16 @@
 @Component
 public class RecipeExecutionFailureCollector {
 
+    public boolean canProcessExecutionFailure(Exception e) {
+        if (e.getCause() != null && e.getCause() instanceof CloudbreakOrchestratorException) {
+            return !((CloudbreakOrchestratorException) e.getCause()).getNodesWithErrors().isEmpty();
+        }
+        if (e.getCause() != null && e.getCause().getCause() != null && e.getCause().getCause() instanceof CloudbreakOrchestratorException) {
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeExecutionFailureCollector.java]: elseif would be better

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeEngine.java (hunk 1 above)
Comment (attilapalfi92) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeEngine.java]: 1) I reordered it (and extracted)
2) It is necessary

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeExecutionFailureCollector.java (hunk 2 above)
Comment (attilapalfi92) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeExecutionFailureCollector.java]: ok

**Pull request merged**